src/pages/patientView/clinicalInformation/lib/clinicalAttributesUtil.js

```diff
@@ -207,21 +207,132 @@ function getSpanElementsFromCleanData(
     clinicalAttributesCleanDerived,
     cancerStudyId
 ) {
-    let spans = [];
-    return Object.keys(clinicalAttributesCleanDerived).map(key => {
-        let value = clinicalAttributesCleanDerived[key];
-        return (
-            <span
-                is
-                class="clinical-attribute"
-                attr-id={key}
-                attr-value={value}
-                study={cancerStudyId}
-            >
-                {value}
-            </span>
-        );
-    });
+    // Order patient clinical attributes
+    const getOrder = key => {
+        const map = {
+            SEX: 1,
+            GENDER: 1,
+            AGE: 2,
+            CANCER_TYPE: 3,
+            CANCER_TYPE_DETAILED: 4,
+            KNOWN_MOLECULAR_CLASSIFIER: 5,
+            HISTOLOGY: 6,
+            OS_STATUS: 7,
+            OS_MONTHS: 8,
+            DFS_STATUS: 9,
+            DFS_MONTHS: 10,
+        };
+
+        return map[key] != undefined ? map[key] : Number.MAX_VALUE;
+    };
+
+    // Ascending order; e.g. GENDER precedes OS_MONTHS
+    const compare = (a, b) => {
+        return getOrder(a) - getOrder(b);
+    };
+
+    // Styles before and after value
+    function stringBuilder(value, key) {
+        var builder = [];
+
+        const before = {
+            /* attributes with opening parenthesis */
+            OS_MONTHS: '\xa0(',
+            DFS_MONTHS: '\xa0(',
+            CANCER_TYPE_DETAILED: '\xa0(',
+            PATIENT_DISPLAY_NAME: '\xa0(',
+            DERIVED_SAMPLE_LOCATION: '\xa0(',
+            SAMPLE_DISPLAY_NAME: '\xa0(',
+
+            /* text before an attribute */
+            SERUM_PSA: ', Serum PSA: ',
+            ERG_FUSION_ACGH: ', ERG-fusion aCGH: ',
+            TMPRSS2_ERG_FUSION_STATUS: ', TMPRSS2-ERG Fusion: ',
+            GLEASON: ', Gleason: ',
+        };
+
+        const after = {
+            /* attributes with a closing parenthesis */
+            CANCER_TYPE_DETAILED: ')',
+            PATIENT_DISPLAY_NAME: ')',
+            DERIVED_SAMPLE_LOCATION: ')',
+            SAMPLE_DISPLAY_NAME: ')',
+
+            /* text after an attribute */
+            OS_MONTHS: ' months)',
+            DFS_MONTHS: ' months)',
+            AGE: ' years old',
+        };
+
+        /* Show comma before clinical attributes */
+        // Prefix is never used, but its assignment allows the side effect of push to happen
+        const prefix =
+            before[key] == '\xa0(' ? builder.push('') : builder.push(', ');
```

Comment:
What? I'm confused. What bug occurs if you do something like this:

```suggestion
        const prefix = before[key] === '\xa0(' ? '' :', ';
        builder.push(prefix);
```

Reply:
That looks nicer, thanks!